### PR TITLE
Update CondDBESSource to match recent EventSetup changes

### PR DESCRIPTION
**Pull request by Dr15Jones**
-Changed to directly inherit from edm::eventsetup::DataProxy since the intermediate DataProxyTemplate was not adding anything useful.
-Switched to modern C++ constructs
-Moved to single memory ownership to reduce use of std::shared_ptr
-Made Initializer be explicitly const

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
could people please review and sign?

**Comment by Dr15Jones** [comment]
Could the L2s please review, this has been sitting here for 2 weeks.

**Comment by lpernie** [comment]
+1

File: CondCore/ESSources/interface/DataProxy.h [hunk 1]
```diff
@@ -125,27 +128,24 @@ class DataProxyWrapper : public  cond::DataProxyWrapperBase {
...
+    m_proxy->setUp( session );
+    m_proxy->loadTag( tag, snapshotTime );
+  }
+
   std::string m_source;
   edm::eventsetup::TypeTag m_type;
-  std::shared_ptr<cond::persistency::PayloadProxy<DataT> >  m_proxy;
-  edmProxyP m_edmProxy;
-
+  std::unique_ptr<PayProxy>  m_proxy;
```
Comment (ggovi) [inline CondCore/ESSources/interface/DataProxy.h]: Any real reason we need this indirection? PayloadProxy already contains a shared_ptr for the payload. Maybe adding full value semantics to PayloadProxy we could remove the unique_ptr?

Comment (Dr15Jones) [reply CondCore/ESSources/interface/DataProxy.h]: `lateInit` gets rid of the old object held by `m_proxy` and makes a new one. That would not be possible with value semantics.

Comment (Dr15Jones) [reply CondCore/ESSources/interface/DataProxy.h]: @ggovi how do you wish to proceed?

Comment (ggovi) [reply CondCore/ESSources/interface/DataProxy.h]: Ok I see! Thanks.

**Comment by ggovi** [comment]
+1

**Comment by fabiocos** [comment]
@thomreis @rekovic could you please check and sign it in case?

**Comment by Dr15Jones** [comment]
@wddgit does this make sense anymore?

**Comment by wddgit** [comment]
I think we should revisit this one after my PR #26592 gets merged and also after a clang format PR for these packages gets merged. Then rebase this one on top of that work. I do not remember this PR at all, but after just a very quick glance it looks like the changes here are improvements ...

**Pull request closed without merging**